### PR TITLE
Disable forms for admins

### PR DESCRIPTION
**Pull request by britt-mo**
## Description

Disable report fields for admin users

### How to test

log in as an admin and see that the report fields are disabled (including nested fields), and report fields for state users are not disabled.

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [ ] ~~I have added analytics, if necessary~~
- [ ] ~~I have updated the documentation, if necessary~~

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/ui-src/src/forms/mcpar/index.ts [hunk 1]
```diff
@@ -3,15 +3,19 @@ import reportSchema from "./mcpar.schema";
 // utils
 import {
   addValidationToReportJson,
+  copyAdminDisabledStatusToForms,
   flattenReportRoutesArray,
 } from "utils/reports/reports";
 import { ReportJson } from "types";
 
+const stateOrAdminVersion: ReportJson =
+  copyAdminDisabledStatusToForms(rawReportJson);
```
Comment (braxex) [inline services/ui-src/src/forms/mcpar/index.ts:12]: Nit: Would maybe name this something like `reportJsonWithDisabledStatus` instead.
```suggestion
const reportJsonWithDisabledStatus: ReportJson =
  copyAdminDisabledStatusToForms(rawReportJson);
```

File: services/ui-src/src/utils/reports/reports.test.ts [hunk 2]
```diff
@@ -112,3 +118,39 @@ describe("Sort Reports", () => {
     expect(sortReportsOldestToNewest(unsortedReports)).toEqual(sortedReports);
   });
 });
+
+describe("Test copyAdminDisabledStatusToForms", () => {
+  const newReportJson: ReportJson = {
```
Comment (braxex) [inline services/ui-src/src/utils/reports/reports.test.ts:123]: We actually have some mockReportJson available [here in setupJest.tsx](https://github.com/CMSgov/mdct-mcr/blob/ddb762731824513e0bd8a78e977c475709a4d036/services/ui-src/src/utils/testing/setupJest.tsx#L230-L235) you could use.

**Review by braxex — CHANGES_REQUESTED** [review]
Looks good! There are a couple things yet to do but we're getting close:

1. Dynamic fields (see form `aap` or `absse`) aren't disabled. This is because the props aren't getting fully passed down into the TextField components that are rendered. Adding `{...props}` to line 40 of `DynamicField` should do the trick.
2. Validation is still running on the forms when admin users click 'Save & continue'. We should stop that from happening. Probably the easiest way is when constructing the form in Form.tsx, we could conditionally tell it if the form needs validation or not by passing the resolver something like this:
```
  const form = useForm({
    resolver: !shouldDisableAllFields ? yupResolver(formSchema) : undefined,
```
3. The 'Save and continue' button probably shouldn't say that anyway. Maybe just 'Continue' if the form is disabled. Would need to modify this in ReportPage.

File: services/ui-src/src/components/forms/Form.tsx [hunk 3]
```diff
@@ -6,15 +6,27 @@ import { yupResolver } from "@hookform/resolvers/yup";
...
   const formSchema = yupSchema(formJson.validation || {});
 
+  // determine if fields should be disabled (based on admin roles )
+  const { userRole } = useUser().user ?? {};
+  const isAdminUser =
+    userRole === UserRoles.ADMIN ||
+    userRole === UserRoles.APPROVER ||
+    userRole === UserRoles.HELP_DESK;
+  const shouldDisableAllFields = isAdminUser && formJson.adminDisabled;
+
```
Comment (braxex) [inline services/ui-src/src/components/forms/Form.tsx:29]: Nit: Would maybe rename this something like `formInputDisabled` or `fieldInputDisabled` or something so when we use it below to determine if a resolver is set it reads a bit better.

**Review by ntsummers1 — APPROVED** [review]
Nice work! I wasn't able to break anything on this and the codes looking good.

**Review by gmrabian — CHANGES_REQUESTED** [review]
Looks like the dynamic field delete button still showed up for admin, and I was able to delete fields. It doesn't actually remove them from the database, and they show up the next time you enter the form. But we should not allow this to be here even so

Screen Shot 2022-09-12 at 8 44 52 AM



Screen Shot 2022-09-12 at 8 44 56 AM

**Review by ntsummers1 — APPROVED** [review]

**Comment by ntsummers1** [comment]
Garretts callout seems fixed. Looks great on my end

**Review by gmrabian — APPROVED** [review]
looks good!

**Review by gmrabian — APPROVED** [review]

**Pull request merged**